tensorboard/plugins/histograms/tf_histogram_dashboard/tf-histogram-loader.html

```diff
@@ -0,0 +1,266 @@
+          if (result.cancelled) {
+            return;
+          }
+          const data = result.value;
+          const formattedData = data.map(this._formatDatum.bind(this));
+          const minmin = d3.min(formattedData, d => d.min);
+          const maxmax = d3.max(formattedData, d => d.max);
+          const normalizedData = formattedData.map(h => ({
+            wall_time: h.wall_time,
+            step: h.step,
+            bins: this._convertBins(h, minmin, maxmax, null),
+          }));
+          this.$$('vz-histogram-timeseries').setSeriesData(
+            this.run, normalizedData);
+        });
+        this.requestManager.request(url).then(updateData);
+      },
+      redraw() {
+        this.$$('vz-histogram-timeseries').redraw();
+      },
+
+      /**
+       * Converts a 1-nested array datum (as returned by the TensorBoard
+       * backend) to a flat object datum (as used by `_convertBins`).
+       */
+      _formatDatum(datum) {
+        const [
+          wall_time,
+          step,
+          [min, max, nItems, sum, sumSquares, bucketRightEdges, bucketCounts],
+        ] = datum;
+        return {
+          wall_time, step, min, max, nItems, sum, sumSquares,
+          bucketRightEdges, bucketCounts,
+        };
+      },
+
+      /**
+       * Takes histogram data as returned by `_formatDatum` and converts
+       * it to the standard d3 histogram data format to make it more
+       * compatible and easier to visualize. When rendering histograms,
+       * having access to the left edge and width of each bin makes
+       * things quite a bit easier, so we include these in the result.
+       * We also convert the bins to have a uniform width, which makes
+       * the visualization easier to understand.
+       *
+       * @param histogram A histogram from tensorboard backend.
+       * @param min The leftmost edge. The binning will start on it.
+       * @param max The rightmost edge. The binning will end on it.
+       * @param numBins The number of bins of the converted data. The
+       * default of 30 is sensible: if you use more, you start to get
+       * artifacts because the event data is stored in buckets, and you
+       * start being able to see the aliased borders between each
+       * bucket.
+       *
+       * TODO(wchargin): This function had some nice tests in
+       * `backendTests.ts`, but these were removed because (a) we moved
+       * this code out of `backend.ts` into its present location and
+       * (b) we weren't running TypeScript tests anyway and. Once we get
+       * TypeScript testing working again, reinstate these tests in a
+       * more appropriate location.
+       *
+       * @return A list of histogram bins. Each bin has an `x` (left
+       *     edge), a `dx` (width), and a `y` (count). If the given
+       *     right edges are inclusive, then these left edges (`x`) are
+       *     exclusive.
+       */
+      _convertBins(histogram, min, max, numBins) {
```

Comment:
Can we:
- Keep this file in Typescript - no need to lose the typing info

- Port the relevant tests to something like histogramTests.ts - even if it doesn't run yet, it's less likely that we'll forget to activate those tests if we have the file lying around, vs. remembering to look at the TODO